Dockerfile.tmpl

```diff
@@ -114,7 +114,7 @@ RUN pip install spacy && \
 
 # b/232247930: uninstall pyarrow to avoid double installation with the GPU specific version.
 RUN pip uninstall -y pyarrow && \
-    mamba install -y cudf cuml && \
+    mamba install -y "pyarrow>=14.0.1" cudf cuml && \
```

Comment:
Let's see if this passes, if it does we'll upgrade both, if not, then we'll switch to only upgrading the CPU-image.

Reply:
Note that *without* this change, it does look like tests pass:
https://ci.kaggle.net/blue/organizations/jenkins/kernels%2Fdocker-python/detail/upgrade-pyarrow/1/pipeline/58/

So cudf is complaining, but it might work anyway...? Though we might not want to risk that.

Reply:
I'd prefer not to risk it.